addons/mail_group/controllers/main.py

```diff
@@ -0,0 +1,266 @@
+from odoo.osv import expression
+
+
+class MailGroup(http.Controller):
+    _thread_per_page = 20
+    _replies_per_page = 5
+
+    def _get_archives(self, group_id):
+        """Return the different date range and message count for the group messages."""
+        def _format_date(datetime_str):
+            """Remove the "hours" part of the datetime and keep only the date part.
+
+            e.g.:
+                "2042-01-01 00:00:00" -> "2042-01-01"
+            """
+            date = datetime.strptime(datetime_str, tools.DEFAULT_SERVER_DATETIME_FORMAT).date()
+            return date.strftime(tools.DEFAULT_SERVER_DATE_FORMAT)
+
+        Message = request.env['mail.group.message']
+
+        results = Message._read_group_raw(
+            [('mail_group_id', '=', group_id)],
+            ['subject', 'create_date'],
+            groupby=['create_date'], orderby='create_date DESC')
+
+        groups = []
+
+        for result in results:
+            (r, label) = result['create_date']
+            start, end = r.split('/')
+
+            groups.append({
+                'date': label,
+                'date_begin': _format_date(start),
+                'date_end': _format_date(end),
+                'messages_count': result['create_date_count'],
+            })
+
+        return groups
+
+    @http.route('/groups', type='http', auth='public', sitemap=True, website=True)
+    def groups(self, email='', **kw):
+        """View of the group lists. Allow the users to subscribe and unsubscribe."""
+        mail_groups = request.env['mail.group'].sudo().search([])
+
+        # Get statistics about the number of messages per month
+        month_date = datetime.today() - relativedelta.relativedelta(months=1)
+        messages_data = request.env['mail.group.message'].read_group([
+            ('create_date', '>=', fields.Datetime.to_string(month_date)),
+            ('moderation_status', '=', 'accepted'),
+        ], ['mail_group_id'], ['mail_group_id'])
+
+        # Dictionary { mail_discusison_id: number_of_message_per_month }
+        messages_data = {
+            message['mail_group_id'][0]: message['mail_group_id_count']
+            for message in messages_data
+        }
+
+        if self._is_user_logged():
+            # Force the email if the user is logged
+            email = request.env.user.email
+
+        # List of group in which the current email is member
+        email_normalized = tools.email_normalize(email)
+        members_data = request.env['mail.group.member'].sudo().search([('email_normalized', '=', email_normalized)])
+        members_data = {member.mail_group_id.id: True for member in members_data}
+
+        return request.render('mail_group.mail_groups', {
+            'mail_groups': mail_groups,
+            'messages_data': messages_data,
+            'members_data': members_data,
+            'email': email,
+        })
+
+    @http.route('/groups/subscription', type='json', auth='public', website=True)
```

Comment:
This "action-based" route is mainly there due to history. Shouldn't we make real route (subscribe and unsubscribe) ? Old route could be kept for backward compatibility **

Reply:
We can, but it will duplicate the code (and add a condition on the JS side)

Reply:
To come back on this point, now in the migration script the group have the same ID of their old mail channel, so the URLs to the portal page will still work, only the token will not work anymore